drivers/flash/flash_simulator.c

```diff
@@ -372,46 +378,58 @@ static int flash_mock_init(const struct device *dev)
 	struct stat f_stat;
 	int rc;
 
-	if (flash_file_path == NULL) {
-		flash_file_path = default_flash_file_path;
-	}
+	ARG_UNUSED(dev);
 
-	flash_fd = open(flash_file_path, O_RDWR | O_CREAT, (mode_t)0600);
-	if (flash_fd == -1) {
-		posix_print_warning("Failed to open flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
-	}
+	if (flash_on_ram == true) {
+		mock_flash = (uint8_t *)malloc(FLASH_SIMULATOR_FLASH_SIZE);
+		if (mock_flash == NULL) {
+			posix_print_warning("Could not allocate flash in Linux process heap %s\n",
+					    strerror(errno));
+			return -EIO;
+		}
+	} else {
 
-	rc = fstat(flash_fd, &f_stat);
-	if (rc) {
-		posix_print_warning("Failed to get status of flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
-	}
+		if (flash_file_path == NULL) {
+			flash_file_path = default_flash_file_path;
```

Comment:
Seems that we can just assign here `DEFAULT_FLASH_FILE_PATH`  and drop the `default_flash_file_path` completely.

Reply:
flash_file_path is just a pointer. So we need something to store that default string.

Reply:
> flash_file_path is just a pointer. So we need something to store that default string.

The `default_flash_file_path` is also just a pointer here.
`DEFAULT_FLASH_FILE_PATH` is name for literal string. You decide its fate by assigning it to something or not. Not assigned is just left out, assigned to `const char *` or `const static char[]` type is placed in ro section of binary, assigned to `static char []` type is placed in writable .data section. Space for it is reserved the moment you reference it.

Reply:
Thank you. Change done.